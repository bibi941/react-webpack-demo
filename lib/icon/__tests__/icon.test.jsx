/*
 * @description:icon.test
 * @author : Fineley.R.Fang(方欣芮)
 * @LastEditors: Fineley.R.Fang(方欣芮)
 * @LastEditTime: 2020-05-27 17:26
*/
import React from 'react'
import renderer from 'react-test-renderer'
import Icon from '../icon'
import { mount } from 'enzyme'

describe('icon', () => {
  it('icon-快照', () => {
    const json = renderer.create(<Icon name={'browse'}/>).toJSON()
    expect(json).toMatchSnapshot()
  })
  it('点击', () => {
    const fn = jest.fn()
    const component = mount(<Icon name={'browse'} onClick={fn}/>)
    component.find('svg').simulate('click')
    expect(fn).toBeCalled
  })
})
