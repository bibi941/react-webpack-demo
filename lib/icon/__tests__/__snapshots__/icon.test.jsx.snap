// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon icon-快照 1`] = `
<svg
  className="ash-icon"
>
  <use
    xlinkHref="#browse"
  />
</svg>
`;
